File: 2024/April/4-9 2073 Time Needed to Buy Tickets/Simulation/TimeNeeded.java
class Solution {
    public int timeRequiredToBuy(int[] tickets, int k) {
        int time = 0;
        int p = 0;
        int length = tickets.length;

        while(tickets[k] > 0) {
            if(tickets[p] > 0) {
                tickets[p] = tickets[p] - 1;
                time++;            
            }

            p++;
            p = p % length;
        }

        return time;
    }
}

File: 2024/April/4-9 2073 Time Needed to Buy Tickets/Simulation/TimeNeeded.py
from typing import List

class Solution:
    def timeRequiredToBuy(self, tickets: List[int], k: int) -> int:
        time = 0
        p = 0
        length = len(tickets)

        while tickets[k] > 0:
            if tickets[p] > 0:
                tickets[p] = tickets[p] - 1
                time += 1
            
            p += 1
            p = p % length
        
        return time